### DISCUSSION TITLE
LocalPeopleResults Displays disabled employees

### DISCUSSION THREAD

**Discussion by jmaszatics25** [microsoft-search/pnp-modern-search#3874] [Q&A] [answered]
Is there a way to exclude inactive employees in the PnP Search based on LocalPeopleResults? Currently my search is displaying disabled accounts (employees that no longer work here). Or is there a way to have it search a Microsoft 365 Group instead?

**Comment by kasperbolarsen** [comment]
Hi, the primary issue is that out of the box people search does not contain a isActive property,  we have to figure it out ourselves. 
I wrote about your options here, https://www.m365thinking.com/post/disabled-or-inactive-user-accounts-in-m365-a-prerequisite-for-governance
Hopes it can assist you

**Reply by jmaszatics25** [reply]
Currently, I am using Sharepoint Search as my available data source for my employee directory. Do I need to switch to Microsoft Search?

**Reply by kasperbolarsen** [reply]
Absolutely not. But you might want to set up a sync job in Azure that will update a property in SP User Profiles based on values from Microsoft Search/Graph

**Reply by jmaszatics25** [reply]
is there a simple way to only have the PnP search use a Microsoft 365 group instead of the active directory for the profiles?

**Reply by kasperbolarsen** [reply]
Good question,  but keeping that Group updated will be hard.
If you can specify the rules for the accounts in that Group you should also be able to trim the LocalPeopleResult, right?
In general I strongly recommend using the HideFromAddressLists as the primary criterion,  as it will also ensure that the account is hidden in Exchange as well.

**Reply by wobba** [reply]
You can delete disabled profiles from the UPA. Or you can set some field on them which you can use to filter them out using KQL. Either way you need som extra job to accomplish one or the other.

**Reply by jmaszatics25** [reply]
I work for a district and I am helping a certain department set up a Sharepoint Page. They requested an employee directory on the page that is updated when an employee enters or leaves the district. I went to my Network admin and asked why employees who no longer work here were displayed in my PnP search of people. I was told that we disable accounts when someone leaves so it must be using our active directory for the localpeople search. My network admin setup a dynamic Microsoft 365 group with only current employees but I have no idea how to get the PnP people search to pull the information from a Microsoft group.

**Reply by kasperbolarsen** [reply]
You should ask the sys admin to set HideFromAddressLists to true when people leave,  that would solve the issue. The Group approach sounds like a bad idea

**Reply by jmaszatics25** [reply]
I just put in a work order ticket and this was their response: "We already set the msExchHideFromAddressLists to True when accounts are disabled.I use a script to process the accounts whenever a staff exits." So even though the person doesn't have an active email they are still included in the PnP Search list.

**Reply by kasperbolarsen** [reply, marked as answer]
In that case, try running this query "SPS-HideFromAddressLists":1
That should list all disabled accounts.  If one of the disabled accounts isn't listed, then the issue is that the AD properties isn't synced correctly to the SharePoint User Profile properties

**Reply by jmaszatics25** [reply]
Where do I run the mentioned query?

**Comment by jmaszatics25** [comment]
Also, is there a way to exclude the profile image?
